lisa/tools/interrupt_inspector.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+import re
+from typing import Dict, List
+
+from lisa.executable import Tool
+from lisa.tools import Cat
+
+
+class Interrupt:
+    irq_number: str
+    interrupt_count: List[int]
+    metadata: str
+    interrupt_sum: int
+
+    def __init__(
+        self,
+        irq_number: str,
+        interrupt_count: List[int],
+        interrupt_sum: int,
+        metadata: str = "",
+    ) -> None:
+        self.irq_number = irq_number
+        self.interrupt_count = interrupt_count
+        self.metadata = metadata
+        self.interrupt_sum = interrupt_sum
+
+    def __str__(self) -> str:
+        return (
+            f"irq_number : {self.irq_number}, "
+            f"count : {self.interrupt_count}, "
+            f"metadata : {self.metadata}"
+            f"sum : {self.interrupt_sum}"
+        )
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+
+class InterruptInspector(Tool):
+    # 0:         22          0  IR-IO-APIC   2-edge      timer
+    _interrupt_regex = re.compile(
+        r"^\s*(?P<irq_number>\S+):\s+(?P<interrupt_count>[\d+ ]+)\s*(?P<metadata>.*)$"
+    )
+
+    @property
+    def command(self) -> str:
+        return "ls -lt /proc/interrupts"
```

Comment:
How about put "cat" here? The cat is actual used command.

Reply:
updated